File: frontend/src/lib/types/admin-media.ts
// Mirrors backend/app/schemas/admin_media.py exactly — keep these two in sync.

export interface MediaFileUpdate {
  alt_text?: string | null;
  title?: string | null;
}

export interface MediaFilePublic {
  id: string;
  file_name: string;
  mime_type: string | null;
  size_bytes: number | null;
  alt_text: string | null;
  title: string | null;
  url: string;
  uploaded_at: string;
}

File: backend/app/schemas/admin_media.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class MediaFileUpdate(BaseModel):
    alt_text: str | None = Field(default=None, max_length=255)
    title: str | None = Field(default=None, max_length=255)


class MediaFilePublic(BaseModel):
    id: uuid.UUID
    file_name: str
    mime_type: str | None
    size_bytes: int | None
    alt_text: str | None
    title: str | None
    url: str
    uploaded_at: datetime

    model_config = {"from_attributes": True}
